conjure-go-client/httpclient/body_types.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2024 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpclient
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/palantir/conjure-go-runtime/v2/conjure-go-contract/codecs"
+)
+
+// RequestBody is an interface that can be used to set the body of an http.Request.
+// Implementations of this interface should set the fields Body, GetBody, and ContentLength (if known).
+type RequestBody interface {
+	setRequestBody(req *http.Request) error
+}
+
+// requestBodyFunc is a function that sets the body of an http.Request and implements RequestBody.
+type requestBodyFunc func() (length int64, body io.ReadCloser, getBody func() (io.ReadCloser, error), err error)
```

Comment:
I think a description like the following would be more helpful:

```suggestion
// requestBodyFunc is a function that returns the length, body reader, and getBody function that should be set on an http.Request. It implements the RequestBody interface by setting the "ContentLength", "Body", and "GetBody" fields of the provided *http.Request to the values returned by the function.
```

Doesn't have to be exactly this, but when I first read the comment and looked at the function signature I was a bit confused because I didn't see how it could set the body of an `http.Request`, so I think it would be helpful for the description to make this a bit clearer.

Reply:
thanks, agreed! updated